parser: add alias for ALTER DATABASE ... SET PRIMARY REGION

otan opened the pull request:
Resolves #59247

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/60603)
<!-- Reviewable:end -->

otan committed:
parser: add alias for ALTER DATABASE ... SET PRIMARY REGION

Resolves #59247

Release note: None

otan committed:
sql: rename SET PRIMARY REGION to PRIMARY REGION as hints

Fix hints where SET PRIMARY REGION is suggested into the more "normal"
form PRIMARY REGION.

Release note: None

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1, 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60603#-:-MTeow1FD9gAI3_QspQi:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm
